### PR TITLE
Code removal

### PR DESCRIPTION
**Pull request by tacaswell**
For some reason I thought this would be 'quick'.  There are still a few more deprecated things that need to be removed, but this is a good start.

We need to make a decision about the future of finance.py.  

I did my best to make sure the commits are atomic so any of them can be dropped now (or reverted later) if need be.

### CODE REVIEW

**Comment by WeatherGod** [comment]
I have changed my mind on removal of qt4_compat.py. I encountered a
conundrum while writing my book. I had to choose between demoing with
qt4_compat.py which has been in mpl since v1.1, but deprecated in v1.4, or
demoing with qt_compat.py, which was introduced in v1.4, so it required the
latest version of matplotlib. I don't think a single release cycle would be
sufficient.

**Comment by tacaswell** [comment]
@WeatherGod removed the removal

File: lib/matplotlib/tri/tripcolor.py [hunk 1]
```diff
@@ -65,6 +64,10 @@ def tripcolor(ax, *args, **kwargs):
     shading = kwargs.pop('shading', 'flat')
     facecolors = kwargs.pop('facecolors', None)
 
+    if shading not in ['flat', 'gouraud']:
+        raise ValueError("shadding must be one of ['flat', 'gouraud'] "
```
Comment (ianthomas23) [inline lib/matplotlib/tri/tripcolor.py]: Typo: 'shadding' -> 'shading'

**Comment by ianthomas23** [comment]
I'm happy with the removal of the deprecated kwarg in tripcolor.

For the delaunay module I had intended keeping it around for a couple of release cycles, but I don't have a particularly strong feeling either way.  When it does finally go we will receive some complaints,  but I don't know that we'll receive more complaints if we remove it sooner rather than later.

**Comment by tacaswell** [comment]
@ianthomas23 I suspect the amount of complaining will be independent of when we do it.  I will defer to you on when it goes.

**Comment by tacaswell** [comment]
@WeatherGod On consideration, your argument is essentially that _all_ re-factoring needs more than one release cycle.

**Comment by WeatherGod** [comment]
It is a bit more nuanced than that. qt4_compat.py and qt_compat.py are
intended to act as compatibility layers. Their entire purpose is to be a
stable platform for developers to avoid the weirdness between different qt
bindings. qt4_compat allows developers to target version 4 of the qt
bindings, while qt_compat allows developers to target version 5+ of the qt
bindings, regardless of which bindings are installed. The compatibility
layers are intended to help developers reach as wide of an audience as
possible. By deprecating qt4 over a single release cycle, you are asking
developers to choose to use a deprecated, but widely used qt4_compat, or
the newly created, not yet widely used qt_compat.py, completely defeating
the purpose of a compatibility layer.

As an example of a refactoring that I have zero problems with taking only a
single release cycle to do: the deprecation of setting the backend via
command-line in favor of an environment variable. I highly doubt that
feature was widely used, nor was it a decent way of doing things in the
first place. It is also fairly trivial to change any scripts to simply not
provide the information via command-line, and instead provide it via an
exported variable. Switching between qt4 and qt5+ is probably not nearly as
trivial.

**Comment by ianthomas23** [comment]
@tacaswell: OK, I think we should keep the delaunay code for one more release cycle before removing it.

**Comment by efiring** [comment]
@tacaswell, Is there a clean way to move this forward?  It looks like most of the removals are non-controversial.  Maybe the thing to do is to move each of the problematic changesets into its own PR, and then merge the remaining majority in the present PR.

**Comment by tacaswell** [comment]
I already have removed the changes that had protests.  I have them as commits on my computer, but haven't gotten around to cherry-picking them on top of current master.

**Comment by QuLogic** [comment]
Here, the `sphinxext/ipython_*` files were removed, but `doc/conf.py` still has a fallback from IPython's to matplotlib's `ipython_console_highlighting` extension. Since that no longer exists, should that fallback also be removed?

**Comment by jenshnielsen** [comment]
@QuLogic Sounds right, will you do a PR?

**Comment by QuLogic** [comment]
I'll put something together as part of #5371.

**Comment by mdboom** [comment]
> I'll put something together as part of #5371.

It looks like this was forgotten before #5371 was merged.  No worries about that, I made #5762...

**Comment by QuLogic** [comment]
Ahh, oops, I meant to say #5743!

**Pull request merged**